gossip/server.go

```diff
@@ -147,8 +152,8 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 			if log.V(1) {
 				log.Infof("node %d received %s from node %d (%d fresh)", s.is.NodeID, extractKeys(args.Delta), args.NodeID, freshCount)
 			}
-			if s.closed {
-				return nil, util.Errorf("gossip server shutdown")
+			if !s.stopper.RunTask(func() {}) {
```

Comment:
A dummy task is weird - the task normally marks out a duration by its begin and end instead of being an instantaneous check. Can we move the rest of this function into the task to ensure that the stopper is not stopped while it is running? (the `closed` boolean was guarded by the mutex but a stopper may be closed by any thread)